File: vm/transaction-builder-generator/examples/rust/stdlib_demo.rs
// Copyright (c) The Diem Core Contributors
// SPDX-License-Identifier: Apache-2.0

use bcs_ext;
use starcoin_stdlib::encode_peer_to_peer_with_metadata_script;
use starcoin_types::{AccountAddress, Identifier, StructTag, TypeTag};

fn main() {
    let token = TypeTag::Struct(StructTag {
        address: AccountAddress([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1]),
        module: Identifier("LBR".into()),
        name: Identifier("LBR".into()),
        type_params: Vec::new(),
    });
    let payee = AccountAddress([
        0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22,
        0x22,
    ]);
    let key_vec = [
        0x22u8, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22,
        0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22, 0x22,
        0x22, 0x22,
    ];
    let amount = 1234567;
    let script = encode_peer_to_peer_with_metadata_script(
        token,
        payee,
        key_vec.to_vec(),
        amount,
        Vec::new(),
    );

    let output = bcs_ext::to_bytes(&script).unwrap();
    for o in output {
        print!("{} ", o);
    }
    println!();
}

File: vm/transaction-builder-generator/examples/python3/stdlib_demo.py
# Copyright (c) The Diem Core Contributors
# SPDX-License-Identifier: Apache-2.0

# pyre-strict

import starcoin_types as starcoin
import serde_types as st
import lcs
import starcoin_stdlib as stdlib


def make_address(content: bytes) -> starcoin.AccountAddress:
    assert len(content) == 16
    # pyre-fixme
    return starcoin.AccountAddress(tuple(st.uint8(x) for x in content))


def main() -> None:
    token = starcoin.TypeTag__Struct(
        starcoin.StructTag(
            address=make_address(b"\x00" * 15 + b"\x01"),
            module=starcoin.Identifier("LBR"),
            name=starcoin.Identifier("LBR"),
            type_params=[],
        )
    )
    payee = make_address(b"\x22" * 16)
    amount = st.uint128(1_234_567)
    script = stdlib.encode_peer_to_peer_with_metadata_script(token, payee, b"",amount, b"")

    for b in lcs.serialize(script, starcoin.Script):
        print("%d " % b, end='')
    print()


if __name__ == "__main__":
    main()
